torch/csrc/jit/mobile/upgrader_mobile.h

```diff
@@ -0,0 +1,74 @@
+#pragma once
+
+// #include <ATen/core/ivalue.h>
+#include <ATen/core/ivalue_inl.h>
+
+#include <string>
+#include <vector>
+#include <torch/csrc/jit/mobile/function.h>
+#include <torch/csrc/jit/serialization/import_export_functions.h>
+#include <unordered_map>
+
+namespace torch {
+namespace jit {
+
+struct Upgrader {
+  int min_version;
+  int max_version;
+  std::string upgrader_name;
+  int index;
+};
+
+// From operator_versions.yaml
+const std::unordered_map<std::string, std::vector<Upgrader>> kOperatorVersionMap(
+    {{std::string("aten::div_Tensor"),
+      std::vector<Upgrader>({Upgrader({0, 3, "div_Tensor_0_3", 0})})}});
+
+
+static std::vector<mobile::Function> kUpgraderFunctions(
+    {
+            mobile::Function::get(
```

Comment:
Could we do something like this:
```
Code code{
{{STOREN, 1, 2}, {LOAD, 1, 0}, ...}, // instructions_
{{"aten::is_floating_point", ""}, {"aten::div", "Tensor"}, {"aten::div", "Tensor_mode"}}, //op_names_
... //constants
...//types
4 // register_size
};
```
It's pretty much direct memory initialization of function. Probably @qihqi could have suggestions on how to initialize a function directly.

It would be very expensive to use tuple and IValue and they may not be necessary.

Reply:
Yes you can create a Function by using it directly: https://github.com/pytorch/pytorch/blob/38c77539e8fbff5b2245e30a0f612efc0c312cae/torch/csrc/jit/mobile/function.h

Basically it takes name as QualifiedName in constructor, then call append_instruction to add instruction and append_operator to add operator.

Reply:
Address in the latest commit.